code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -39,14 +39,14 @@
 
 /obj/item/weapon/melee/baton/proc/deductcharge(chrgdeductamt)
 	if(bcell)
+		//Note this value returned is significant, as it will determine
+		//if a stun is applied or not
 		. = bcell.use(chrgdeductamt)
-		if(bcell.charge >= hitcost) // If after the deduction the baton doesn't have enough charge for a stun hit it turns off.
-			return
-	if(status)
-		status = 0
-		update_icon()
-		playsound(loc, "sparks", 75, 1, -1)
-	return 0

```

Comment:
this was the bug, as it returned false, instead of returning the actual value of the bcell.use call